src/Common/src/System/Collections/Generic/LargeArrayBuilder.cs

```diff
@@ -216,25 +216,36 @@ public CopyPosition CopyTo(CopyPosition position, T[] array, int arrayIndex, int
             int row = position.Row;
             int column = position.Column;
 
-            for (; count > 0; row++, column = 0)
+            T[] buffer = GetBuffer(row);
+            int copied = CopyToCore(buffer, column);
+
+            while (count > 0)
+            {
+                buffer = GetBuffer(++row);
+                copied = CopyToCore(buffer, 0);
+            }
+
+            return copied == buffer.Length
+                ? new CopyPosition(row + 1, 0)
+                : new CopyPosition(row, copied);
+
+            int CopyToCore(T[] sourceBuffer, int sourceIndex)
             {
-                T[] buffer = GetBuffer(index: row);
+                Debug.Assert(sourceBuffer.Length > sourceIndex);
```

Comment:
You mentioned previously this assert failed. It's good to see that's no longer the case.

Reply:
Yes, because of the `if (count > 0)` check added to the calling method.